### PR TITLE
Re-organize data passing for flexibility

### PR DESCRIPTION
**Pull request by unframework**

### CODE REVIEW

**Comment by atesgoral** [comment]
👍

File: angular-action.js [hunk 1]
```diff
@@ -15,98 +15,180 @@
...
+        var fieldDeferredValueMap = {};
+
+        ownerScope.$on('$actionDataObjectFieldCreated', function (event, name) {
+            // consume the data response
+            event.stopPropagation();
+
+            var fieldScope = event.targetScope;
+
+            if (fieldScopeMap[name]) {
+                // @todo might be over-zealous in case of dynamic field replacement
```
Comment (paracycle) [inline angular-action.js]: Agreed. We do lots of `ng-if`s on individual input elements that may get created and destroyed multiple times. If we are doing this, we should at least have a corresponding `$actionDataObjectFieldDestroyed` event that clears the slot.

Comment (paracycle) [inline angular-action.js]: Oh wait, I guess we are doing the cleanup in the .$on('$destroy')` below. Strike that comment.

**Comment by paracycle** [comment]
👍

**Pull request merged**